traffic_ops/traffic_ops_golang/parameter/parameters.go

```diff
@@ -176,10 +176,14 @@ func (param *TOParameter) Read(h http.Header, useIMS bool) ([]interface{}, error
 			return nil, nil, errors.New("scanning " + param.GetType() + ": " + err.Error()), http.StatusInternalServerError, nil
 		}
 		if p.Secure != nil && *p.Secure {
-			if param.ReqInfo.Version.Major >= 4 &&
-				param.ReqInfo.Config.RoleBasedPermissions &&
-				!param.ReqInfo.User.Can("PARAMETER-SECURE:READ") {
-				p.Value = &HiddenField
+			if param.ReqInfo.Version.Major >= 4 {
+				if param.ReqInfo.Config.RoleBasedPermissions {
+					if !param.ReqInfo.User.Can("PARAMETER-SECURE:READ") {
+						p.Value = &HiddenField
+					}
+				} else if param.ReqInfo.User.PrivLevel < auth.PrivLevelAdmin {
+					p.Value = &HiddenField
+				}
```

Comment:
So in APIv5 and later we want to not consider "Priv Level" no matter what. The configuration file option that does that is not meant to act on APIv5 - but it currently does. So far that's fine because APIv5 isn't released, but it's something we need to fix. The question, then, is do you want to do that here right now, or leave it consistent with the rest of the server's behavior and then when that's fixed someone will need to remember to fix this, too?

Reply:
By fix do you mean adding a conditional for version >= 5 to not check the config and only use permissions?

Reply:
yeah, exactly